kobo/settings/base.py

```diff
@@ -819,9 +819,12 @@ def __init__(self, *args, **kwargs):
         MONGO_DB_URL = "mongodb://%(HOST)s:%(PORT)s/%(NAME)s" % MONGO_DATABASE
     mongo_db_name = MONGO_DATABASE['NAME']
 else:
-    # Get collection name from the connection string, fallback on 'formhub' if
-    # it is empty or None
-    mongo_db_name = env.db_url('MONGO_DB_URL').get('NAME') or 'formhub'
+    # Attempt to get collection name from the connection string
+    # fallback on KPI_MONGO_NAME or 'formhub' if it is empty or None or unable to parse
+    try:
+        mongo_db_name = env.db_url('MONGO_DB_URL').get('NAME') or 'formhub'
+    except ValueError: # db_url is unable to parse replica set strings
+        mongo_db_name = env.str('KPI_MONGO_NAME', 'formhub')
```

Comment:
The warning on line 794 would not be accurate anymore 

> `MONGO_DB_URL` is not found. 
>`KPI_MONGO_HOST`, `KPI_MONGO_PORT`, `KPI_MONGO_NAME`, `KPI_MONGO_USER`, `KPI_MONGO_PASS` are deprecated and will not be supported anymore soon.

I would introduce a new variable `MONGO_DB_NAME` instead. Moreover, `MONGO_DB_URL` remove the concept of two different MongoDB connections (one for KPI , one for KoBoCAT). 

Long story short, I would keep `KPI_MONGO_NAME` deprecated and would use `MONGO_DB_NAME` instead.
What about something like:

```diff
-     # Get collection name from the connection string, fallback on 'formhub' if
-     # it is empty or None
-     mongo_db_name = env.db_url('MONGO_DB_URL').get('NAME') or 'formhub'
+    # Attempt to get collection name from the connection string
+    # fallback on KPI_MONGO_NAME or 'formhub' if it is empty or None or unable to parse
+    try:
+        mongo_db_name = env.db_url('MONGO_DB_URL').get('NAME') or env.str('MONGO_DB_NAME', 'formhub')
+    except ValueError: # db_url is unable to parse replica set strings
+        mongo_db_name = env.str('MONGO_DB_NAME', 'formhub')
```

Notice I've also replaced `or 'formhub'` with `or env.str('MONGO_DB_NAME', 'formhub')` on line 825.

Reply:
Slight code golf `mongo_db_name = env.db_url('MONGO_DB_URL').get('NAME', env.str('MONGO_DB_NAME', 'formhub'))` since the second param is the default.

Reply:
> So the preferred env vars that we will document are `MONGO_DB_URL` and `MONGO_DB_NAME`, right? Anything else is legacy.

Indeed

Reply:
I think we should keep the other alternative.

Let's use as an example: 
```python 
MONGO_DB_URL = 'mongodb://user:password@mongo.kobo.tld:27017'  # no collection provided
```

```python 
In [1]: env.db_url('MONGO_DB_URL').get('NAME') or env.str('MONGO_DB_NAME', 'formhub')
In [1]: formhub

In [2]: env.db_url('MONGO_DB_URL').get('NAME', env.str('MONGO_DB_NAME', 'formhub'))
Out[2]: ''
```

Because `NAME` (always) exists in the dict the `db_url('MONGO_DB_URL')` returns. It's an empty string, thus get('NAME', ...) does not fallback on its default.

Reply:
Updated and rebased